src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRequestExecutor.java

```diff
@@ -181,6 +192,14 @@ private <T> BitbucketResponse<T> makeGetRequest(HttpUrl url, ObjectReader<T> rea
                 }, addCredentials(additionalConfig));
     }
 
+    private InputStream streamGetRequest(HttpUrl url, RequestConfiguration... additionalConfig) {
+        return new ByteArrayInputStream(httpRequestExecutor.executeGet(url,
+                response -> {
+                    ensureNonEmptyBody(response);
+                    return response.body().bytes();
+                }, addCredentials(additionalConfig)));
+    }
```

Comment:
Not keen on wrapping the response into a `ByteArrayInputStream` if we can avoid it. Can we not "just" do

```java
 return httpRequestExecutor.executeGet(url,
                response -> {
                    ensureNonEmptyBody(response);
                    return response.body().byteStream();
                }, addCredentials(additionalConfig));
    }
    
```
?

Reply:
I tried playing around with that, however the stream is closed once we leave the lambda function. This way we are essentially creating a copy of the stream that is usable by the caller. It's not ideal but its the best i could come up with.